test/e2e/util.go

```diff
@@ -2141,3 +2141,36 @@ func OpenWebSocketForURL(url *url.URL, config *client.Config, protocols []string
 	cfg.Protocol = protocols
 	return websocket.DialConfig(cfg)
 }
+
+// getIngressAddress returns the ips/hostnames associated with the Ingress.
+func getIngressAddress(client *client.Client, ns, name string) ([]string, error) {
+	ing, err := client.Extensions().Ingress(ns).Get(name)
+	if err != nil {
+		return nil, err
+	}
+	addresses := []string{}
+	for _, a := range ing.Status.LoadBalancer.Ingress {
+		if a.IP != "" {
+			addresses = append(addresses, a.IP)
+		}
+		if a.Hostname != "" {
+			addresses = append(addresses, a.Hostname)
+		}
+	}
+	return addresses, nil
+}
+
+// waitForIngressAddress waits for the Ingress to acquire an address.
```

Comment:
Someone reading the function header would have no idea what the string return value would contain, unless they read through the code.

What I was suggesting is one of the following instead, approximately, in increasing order of my preference:

```
func waitForIngressAddress(c *client.Client, ns, ingName string, timeout time.Duration) (address string, error)
```

```
type address string
...
func waitForIngressAddress(c *client.Client, ns, ingName string, timeout time.Duration) (address, error)
```